packages/strapi-admin/admin/src/containers/AdminPage/index.js

```diff
@@ -37,15 +40,43 @@ export class AdminPage extends React.Component { // eslint-disable-line react/pr
       plugins: this.props.plugins,
       updatePlugin: this.props.updatePlugin,
     }
-  )
+  );
+
+  componentDidMount() {
+    if (this.hasUsersPlugin() && this.isUrlProtected(this.props) && !auth.getToken()) {
```

Comment:
Duplicated code, maybe put it in a tiny function it will be more maintainable ;)